babel/src/test/resources/sql/big-query.iq

```diff
@@ -2519,4 +2519,31 @@ SELECT PARSE_TIMESTAMP("%c", "Thu Dec 25 07:30:00 2008") AS parsed;
 !ok
 !}
 
+#####################################################################
+#
+# STARTS_WITH(value1, value2)
+#
+# Takes two STRING or BYTES values. Returns TRUE if the second value is a prefix of the first.
+#
+# This function supports specifying collation.
+WITH items AS
+  (SELECT 'foo' as item
```

Comment:
These are all strings, but your function accepts `OperandTypes.STRING_STRING` which includes bytes objects as well. We should test those too (I think you need another overload in `SqlFunctions`).

I also think you may need to specify that both arguments must have the same type, and I'm not sure if `OperandTypes.STRING_STRING` accomplishes that. There should be a test to make sure you can't call `STARTS_WITH` passing a string and a bytes together. Double check that this is consistent with BigQuery by running a sample query in the BQ console, since their documentation isn't 100% clear.

Reply:
Update: BQ Console will throw an error if attempting to use string+bytes, so I've updated this PR to use `STRING_SAME_SAME`